dask_cuda/device_host_file.py

```diff
@@ -10,14 +10,32 @@
 def _is_device_object(obj):
     """
     Check if obj is a device object, by checking if it has a
-    __cuda_array_interface__ attributed
+    __cuda_array_interface__ attribute or is instance of cudf.DataFrame
+    or cudf.Series.
     """
-    return hasattr(obj, "__cuda_array_interface__")
+    try:
+        import cudf
+        _device_instances = [cudf.DataFrame, cudf.Series]
+    except ImportError:
+        _device_instances = []
+    return (hasattr(obj, "__cuda_array_interface__") or
+            any([isinstance(obj, inst) for inst in _device_instances]))
+
+
+def _is_device_object_multi(obj):
+    """
+    Check if obj is a device object with _is_device_object(). If obj
+    is a list or tuple, check if at least one of them is a device object.
+    """
+    if isinstance(obj, list) or isinstance(obj, tuple):
+        return any([_is_device_object(o) for o in obj])
+    else:
+        return _is_device_object(obj)
```

Comment:
Maybe roll this logic into `_is_device_object` directly?

Reply:
Ping

Reply:
Since `_is_device_object` is not just a single condition now, I'd rather keep it separate than duplicating that code.

Reply:
Perhaps some light recursion:

```python
def is_device_object(obj):
    if isinstance(obj, (list, tuple)):
        return any(map(is_device_object, obj))
    elif ...
```